src/main/java/com/hartwig/platinum/kubernetes/TargetNodePool.java

```diff
@@ -0,0 +1,63 @@
+package com.hartwig.platinum.kubernetes;
+
+import com.hartwig.platinum.config.NodePoolConfiguration;
+import com.hartwig.platinum.config.SampleConfiguration;
+
+public interface TargetNodePool {
+
+    boolean isDefault();
+
+    String name();
+
+    String machineType();
+
+    int numNodes();
+
+    static TargetNodePool fromConfig(final NodePoolConfiguration configuration, final int samples) {
+        return new TargetNodePool() {
+            @Override
+            public boolean isDefault() {
+                return false;
+            }
+
+            @Override
+            public String name() {
+                return configuration.name();
+            }
+
+            @Override
+            public String machineType() {
+                return "e2-medium";
+            }
+
+            @Override
+            public int numNodes() {
+                return Math.max(1, samples / 10);
+            }
+        };
+    }
+
+    static TargetNodePool defaultPool() {
```

Comment:
Unless I'm misunderstanding, the default pool can't actually be used for anything because of the `UnsupportedOperationException`s. If this is the case I feel as though we could be more explicit in the naming, something like `unspecifiedDefaultPool`. Dunno, maybe this is just a nit.

Reply:
the `isDefault` method returning true has the impact of using the default node pool of the cluster, which does not require a name or machine type.